lib: otdoa_al: force config dl in ubsa request

serranozack opened the pull request:
Adds a parameter to ubsa download requests to force a config file download along with it

serranozack committed:
lib: otdoa_al: force config dl in ubsa request

Adds a parameter to ubsa download requests to force a config file
download along with it

Signed-off-by: Zack Mudd <zmudd@serranosystems.com>

nriedel commented on lib/otdoa_al/otdoa_http_api.c:
```diff
@@ -112,6 +112,15 @@ int32_t otdoa_api_ubsa_download(const otdoa_api_ubsa_dl_req_t *p_dl_request,
+
+	/* if a config file download is being forced, get that first */
+	if (p_dl_request->config_dl) {
+		rc = otdoa_api_cfg_download();
```
I think you need to set a flag in the `tOTDOA_MSG_HTTP_GET_UBSA` message to request a config file also be downloaded.  With what you have here, two messages are sent to the workqueue thread (one for config DL and one for uBSA DL).  If the uBSA DL fails, we will still try to DL the uBSA since it is triggered by a second message.  This can result in all kinds of confusion.  In addition, there is the DL the config file every N uBSA DLs.  That should be reset if the config DL is forced.

nriedel reviewed (changes requested):
Just the one comment here.  It probably will require a little bit of rework.

serranozack committed:
lib: otdoa_al: config file dl trigger in ubsa dl

Moves handling for automatic and forced config file downloads from the
uBSA download API to inside the uBSA download message handler,
preventing duplicate messages from being sent.

Signed-off-by: Zack Mudd <zmudd@serranosystems.com>

serranozack committed:
lib: otdoa_al: fix compliance

C99 comments are not allowed and parameterless functions must take void

Signed-off-by: Zack Mudd <zmudd@serranosystems.com>
